app/code/Magento/Sales/Block/Order/Totals.php

```diff
@@ -198,7 +198,10 @@ public function addTotal(\Magento\Framework\DataObject $total, $after = null)
             }
             if (!$added) {
                 $last = array_pop($totals);
-                $totals[$total->getCode()] = $total;
+                // Push tax before grand total.
+                $totals =  array_slice($totals, 0, 2, true) +
+                    [$total->getCode() => $total] +
+                    array_slice($totals, 2, count($totals) -1, true);
```

Comment:
There is no need to change method, instead of this modify call so that tax  before grand total is explicit.